warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/BulkIngestMapFileLoader.java

```diff
@@ -527,13 +528,29 @@ public void run() {
                     log.error("Error: " + e.getMessage(), e);
                 }
             }
+            
         } finally {
             log.info("Shutting down executor service");
             executor.shutdown();
         }
         log.info("Bulk map file loader shutting down.");
     }
     
+    protected void cleanJobDirectoriesOnStartup() throws IOException {
+        Path[] cleanupDirectories = getJobDirectories(destHdfs, new Path(workDir, jobDirPattern + '/' + CLEANUP_FILE_MARKER));
+        for (int i = 0; i < cleanupDirectories.length; i++) {
+            
+            markSourceFilesLoaded(cleanupDirectories[i]);
+            try {
+                getFileSystem(destHdfs).delete(cleanupDirectories[i], true);
+            } catch (IOException e) {
+                log.warn("Unable to delete directory " + cleanupDirectories[i] + ".  Another process already cleaned it up.");
```

Comment:
How do you know that is the reason for the IOException?  Perhaps the client could not reach the NN at the time or something else.  Include the message from the IOException as the reason.

Reply:
updated.